.grit/patterns/ternary_op.md

```diff
@@ -0,0 +1,67 @@
+---
+title: Replace if-else with ternary operation where appropriate
+---
+
+Replaces an assignment to the same variable done across an if-else with a ternary operator when both are equivalent.
+
+```grit
+engine marzano(0.1)
+language python
+
+// Helper to check if a token is a const literal
+pattern is_const_literal() {
+    r"(?:\d+.\d+)|(?:\d+)|(?:'.*')|(?:\".*\")$"
+}
+
+`
+if $cond:
+    $if_body
+else:
+    $else_body
+` as $cond_body where {
+    $if_body <: contains `$var = $if_value` && $else_body <: contains `$var = $else_value` where and {
+        $if_value <: is_const_literal(),
+        $else_value <: is_const_literal(),
+        $cond_body => `$var = $if_value if $cond else $else_value`,
+    },
```

Comment:
We are deprecating &&. (use `and` instead.)

Also I'm not sure contains is correct. What if there are other side effects?

Add a sample covering those.

Reply:
So I split this case into two.
The first case would be:
```python
if cond:
    x = some_call()
else:
    x = other_call()
```
The second case would be:
```python
if cond:
    y = 10
    x = 1
else:
    x = 2
```
In the second case we cannot be sure that the assignment to x is not from some data created in the if-else body. So we only rewrite const assignment.